src/internal/components/button-trigger/index.tsx

```diff
@@ -99,7 +99,7 @@ const ButtonTrigger = (
       {children}
       {!hideCaret && (
         <span className={styles.arrow}>
-          <InternalIcon name="caret-down-filled" variant={disabled ? 'disabled' : 'normal'} />
+          <InternalIcon name="caret-down-filled" variant={disabled || readOnly ? 'disabled' : 'normal'} />
```

Comment:
Didn't introduce a new variant because the `InternalIcon` doesn't use any `aria-*` attributes, and `disabled` and `readOnly` share the same style rules